src/metrics/metrics.js

```diff
@@ -0,0 +1,40 @@
+import config from 'config';
+
+import log from '../log';
+import console from './console';
+import influx from './influx';
+import timingMetric from './timingMetric';
+
+export const VALID_TYPES = ['request', 'redirect', 'generation', 'uploadToCache', 'upload', 'original'];
```

Comment:
It seems to me that this restriction is rather arbitrary, and mostly intended to catch typos. I would suggest to instead expose these values as an enum-like object, and omit the validation on them.